### PR TITLE
Maven property hovers should MarkupContent use "markdown" kind

### PR DESCRIPTION
**Pull request by AObuchow**
Part of #97

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

### CODE REVIEW

**Comment by mickaelistria** [comment]
I don't think it's correct. The Maven parameter documentation is not markdown, so it shouldn't be marked as being markdown.
It could be a bug in LSP4E not properly rendering the documentation in case it's not markdown; but I don't think the LS should be where we try to hack things.

**Comment by AObuchow** [comment]
@mickaelistria the `MavenPluginUtils.LINE_BREAK` which is used in the maven property hover is `\n\n` which is markdown. I do think that an investigation in LSP4E needs to be made, however, this patch seems correct to me.

**Comment by mickaelistria** [comment]
I think using markdown is strictly incorrect as Maven doc seems to say that documentations are typically using Javadoc format. However, as Javadoc is partly HTML and Markdown renderer is supposed to partly render HTML, we can make the switch.
ie it's incorrect, but less incorrect than plaintext.

File: lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenHoverParticipant.java [hunk 1]
```diff
@@ -327,7 +327,7 @@ private Hover collectProperty(IPositionRequest request, String property) {
 			for (Entry<String, String> prop : allProps.entrySet()) {
 				String mavenProperty = prop.getKey();
 				if (property.equals(mavenProperty)) {
-					return new Hover(new MarkupContent("plaintext", "Property: " + mavenProperty
+					return new Hover(new MarkupContent("markdown", "Property: " + mavenProperty
```
Comment (angelozerr) [inline lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenHoverParticipant.java:330]: You should take care if client can support markdown.

To do that:

`boolean markdown = request.canSupportMarkupKind(MarkupKind.MARKDOWN);`

After that I suggest you that you do like we have done for entites doc (for hover and completion). See https://github.com/eclipse/lemminx/blob/374ea32776f24a910f6a2541d0f66dd21cff1266/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/entities/EntitiesDocumentationUtils.java#L91

**Comment by mickaelistria** [comment]
ah sorry, I just realized that your patch is about _properties_ while I had in mind _parameters_. So yes, as the properties don't have a specific documentation, it's fine to use markdown.

File: lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenHoverParticipant.java (hunk 1 above)
Comment (mickaelistria) [reply lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenHoverParticipant.java:330]: At the moment, I think we can just not care too much about such clients as they should be marginal. We'll work on that when we have a request for it from some integrator and we consider it's high priority. Realistically, it's more the kind of things we will let the people who want it implement.

**Comment by angelozerr** [comment]
> I think using markdown is strictly incorrect as Maven doc seems to say that documentations are typically using Javadoc format. However, as Javadoc is partly HTML and Markdown renderer is supposed to partly render HTML, we can make the switch.
ie it's incorrect, but less incorrect than plaintext.

We had the same problem with XSD documentation. To fix that we generate HTML content and convert the HTML to markdown.

I suggest you that you generate your doc in HTML and after that you consume https://github.com/eclipse/lemminx/blob/374ea32776f24a910f6a2541d0f66dd21cff1266/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/MarkupContentFactory.java#L41

File: lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenHoverParticipant.java (hunk 1 above)
Comment (angelozerr) [reply lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenHoverParticipant.java:330]: If you use https://github.com/eclipse/lemminx/blob/374ea32776f24a910f6a2541d0f66dd21cff1266/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/MarkupContentFactory.java#L41 everything should be supported

**Comment by AObuchow** [comment]
Thank you for the advice @angelozerr I like that approach. As we are creating markdown in a few areas of the code, I feel that a separate bug should be made to use this approach everywhere in LemMinX-Maven (not just in this commit). I'll go write that now :)

**Pull request merged**